### DISCUSSION TITLE
How to use the CppUTestBuildTimeDiscoverTests.cmake script

### DISCUSSION THREAD

**Discussion by DBizz22** [cpputest/cpputest#1817] [Q&A]
Hello, Goodday. Please, how does one use the cpputest_buildtime_discover_tests function in the CCppUTestBuildTimeDiscoverTests.cmake script that comes with msys2 CppUTest package. I thought it would work like the cpputest_discover_tests function that comes with the standard project but ctest is having difficult finding the tests

**Comment by thetic** [comment]
I'm not sure the exact version of CppUTest is packaged by msys. Assuming it's close to the last release 4.0, you should be able to something like

```cmake
cmake_minimum_required(VERSION 3.1)
project(example_project)

# this enables CTest for the project.
include(CTest) # or enable_testing()

find_package(CppUTest REQUIRED)

add_executable(example_test #[[ ... ]] )
target_link_libraries(example_test PRIVATE CppUTest)
cpputest_buildtime_discover_tests(example_test)
```

`cpputest_buildtime_discover_tests` should function similarly to the more recent `cpputest_discover_tests`, but may not support as many features or edge cases.

**Reply by DBizz22** [reply]
MSYS2 uses version 4.0-2, it seems to build the test executable file fine but features like being able to discover the tests by ctest from the top level directory is not working

**Reply by thetic** [reply]
Please explain the specific behavior you are seeing including any logs.

**Reply by DBizz22** [reply]
<img width="1001" alt="terminal4" src="https://github.com/user-attachments/assets/5fc35e6b-a747-4a2f-9e52-3dd0224c737f">

ctest can not discover the tests built using the Msys2 package, i suspect due to the cpputest_buildtime_discover_tests function,  but when i tried building the same test using cpputest_discover_tests and targets from the github package, ctest could discover the tests

**Reply by thetic** [reply]
If you remove the call to `cpputest_buildtime_discover_tests`, does running the compiled executable with the `-lg` option (list groups) give the expected output?

**Reply by DBizz22** [reply]
> If you remove the call to `cpputest_buildtime_discover_tests`, does running the compiled executable with the `-lg` option (list groups) give the expected output?

Yes, it gives the list of the test groups
